src/common/serializer/deserializer.cpp

```diff
@@ -0,0 +1,15 @@
+#include "common/serializer/deserializer.h"
+
+namespace kuzu {
+namespace common {
+
+template<>
+void Deserializer::deserializeValue<std::string>(std::string& value) {
```

Comment:
I think you don't need to write the specialization explicitly in the implementation, this should work:
`void Deserializer::deserializeValue(std::string& value)`